### PR TITLE
ISAICP-5939: New author role for trusted content authorship.

### PR DESCRIPTION
**Pull request by idimopoulos**

### CODE REVIEW

File: config/sync/core.entity_view_display.rdf_entity.collection.about.yml [hunk 1]
```diff
@@ -139,17 +140,15 @@ content:
...
       format: default
-      format_custom_true: ''
       format_custom_false: ''
-    third_party_settings:
-      template_suggestion:
-        template_suggestion: fieldset
-    type: boolean
-    region: content
+      format_custom_true: ''
+    third_party_settings: {  }
```
Comment (pfrenssen) [inline config/sync/core.entity_view_display.rdf_entity.collection.about.yml:151]: These changes are causing a regression in the about page:

moderation-label

The label is now shown, but this duplicates the label that is already there.

File: config/sync/field.field.rdf_entity.collection.field_ar_moderation.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ id: rdf_entity.collection.field_ar_moderation
 field_name: field_ar_moderation
 entity_type: rdf_entity
 bundle: collection
-label: Moderation
+label: Moderated
```
Comment (pfrenssen) [inline config/sync/field.field.rdf_entity.collection.field_ar_moderation.yml:14]: Ah I might be mistaken in my comment above, I possibly mixed up my screenshots. I agree that this should be "Moderated" and not "Moderation".

File: config/sync/og.og_role.rdf_entity-collection-author.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  - 'delete own event content'
+  - 'delete own news content'
+  - 'delete own news content'
+  - 'edit own custom_page content'
+  - 'edit own discussion content'
+  - 'edit own document content'
+  - 'edit own event content'
+  - 'edit own news content'
+  - 'post comments'
+  - 'view own unpublished content'
```
Comment (pfrenssen) [inline config/sync/og.og_role.rdf_entity-collection-author.yml:33]: All of these permissions are already present in the standard member role, so they should not be duplicated here.

File: web/modules/custom/joinup_community_content/joinup_community_content.module [hunk 4]
```diff
@@ -133,6 +133,14 @@ function joinup_community_content_workflow_selector(EntityInterface $entity): ?s
     return NULL;
   }
 
+  $user_id = \Drupal::currentUser()->id();
+  /** @var \Drupal\og\OgMembershipInterface $membership */
+  if ($membership = \Drupal::getContainer()->get('og.membership_manager')->getMembership($parent_entity, $user_id)) {
+    if ($membership->hasRole($parent_entity->getEntityTypeId() . '-' . $parent_entity->bundle() . '-author')) {
+      return 'node:post_moderated';
+    }
+  }
```
Comment (pfrenssen) [inline web/modules/custom/joinup_community_content/joinup_community_content.module:142]: This is hacky so it is a good idea to explain why it is necessary that we mess with the post_moderated state depending on the membership.

File: web/modules/custom/joinup_core/joinup_core.post_update.php [hunk 5]
```diff
@@ -48,3 +48,38 @@ function joinup_core_post_update_0106400(): void {
...
+INSERT { ?entity_id <http://joinup.eu/collection/new_role> "rdf_entity-collection-author" }
+WHERE { ?entity_id <http://joinup.eu/collection/moderation> "false"^^<http://www.w3.org/2001/XMLSchema#boolean> }
+QUERY;
+
+    $connection->query($enabled);
+    $connection->query($disabled);
+  }
+
+  \Drupal::entityTypeManager()->getStorage('rdf_entity')->resetCache($ids);
+}
```
Comment (pfrenssen) [inline web/modules/custom/joinup_core/joinup_core.post_update.php:85]: This is not clear to me, can you explain why this is needed? The non-moderated collections don't need the author role added to function like expected so this update hook seems to be useless.

It is also not asked in the ticket that we should assign the author role to all members of collections that are not moderated.

**Review by pfrenssen — CHANGES_REQUESTED** [review]

File: web/modules/custom/joinup_community_content/joinup_community_content.module (hunk 4 above)
Comment (idimopoulos) [reply web/modules/custom/joinup_community_content/joinup_community_content.module]: I have added a commend though this is not really hacky. This is the whole meaning of the ticket. Having a role that behaves as a post moderated role in any group.

**Comment by pfrenssen** [comment]
This is OK if tests come back green.

**Review by pfrenssen — APPROVED** [review]

**Review by pfrenssen — APPROVED** [review]

**Pull request merged**